torch/testing/_internal/common_methods_invocations.py

```diff
@@ -799,6 +850,38 @@ def sample_inputs(self, device, dtype, requires_grad=False):
                                 dtypes=[torch.cfloat, torch.cdouble])),
                    promotes_integers_to_float=True,
                    handles_complex_extremals=False),
+    OpInfo('linalg.pinv',
+           aten_name='linalg_pinv',
+           op=torch.linalg.pinv,
+           dtypes=floating_and_complex_types(),
+           test_inplace_grad=False,
+           supports_tensor_out=False,
+           sample_inputs_func=sample_inputs_linalg_pinv,
+           decorators=[skipCUDAIfNoMagma, skipCPUIfNoLapack],
+           skips=(
+               # TODO: RuntimeError: svd does not support automatic differentiation for outputs with complex dtype.
+               # TODO: Add complex types here
+               # See https://github.com/pytorch/pytorch/pull/47761
+               SkipInfo('TestGradients', 'test_fn_grad', dtypes=[torch.cdouble]),
+               SkipInfo('TestGradients', 'test_fn_gradgrad', dtypes=[torch.cdouble]),
+               SkipInfo('TestGradients', 'test_method_grad', dtypes=[torch.cdouble]),
+               SkipInfo('TestGradients', 'test_method_gradgrad', dtypes=[torch.cdouble]),
+               SkipInfo('TestCommon', 'test_variant_consistency_eager', dtypes=[torch.cfloat, torch.cdouble]),
+               SkipInfo('TestCommon', 'test_variant_consistency_jit', dtypes=[torch.cfloat, torch.cdouble])),
+           ),
+    HermitianOpInfo('linalg.pinv',
+                    variant_test_name='hermitian',
+                    aten_name='linalg_pinv',
+                    op=torch.linalg.pinv,
+                    dtypes=floating_and_complex_types(),
+                    test_inplace_grad=False,
+                    supports_tensor_out=False,
+                    sample_inputs_func=sample_inputs_linalg_pinv_hermitian,
+                    decorators=[skipCUDAIfNoMagma, skipCPUIfNoLapack],
+                    skips=(
+                        # These tests do not take into account custom op.get_op()
```

Comment:
Can you elaborate on this? It seems like there should be a simple fix for it. For example, should the call dunder on HermitianOpInfo be updated to behavior similarly to its get_op?

Reply:
For gradchecks to pass for Hermitian functions (`cholesky, eigh, pinv(hermitian=True)`) a [modified function](https://github.com/pytorch/pytorch/pull/48399/files#diff-18a0c3e7cd07089b34760028c387d39de1aa393172f1d5f123692106b3a3d144R387-R389) must be used.
These JIT tests do not use `op.get_op()` for gradchecks, so they fail. From a quick glance, I didn't see how to fix it properly.

Reply:
I see. We had a similar issue with a recent PR creating OpInfos for functions like stack. The current version of gradcheck doesn't handle iterables of tensors as arguments, so we ended up making the op for stack be a wrapper around torch.stack that took a variable number of arguments and then passed them as an iterable.

In that case we were lucky that this new version of the op could work for both gradchecks and the regular function checks performed on the operation, but it was a hack around gradcheck's limitations.

Would adding something like "grad_op" to the OpInfo be the right approach here? That could also be used for cases like torch.stack.

Reply:
Here the problem is that [in `test_variant_consistency_jit` from `test/test_ops.py`](https://github.com/pytorch/pytorch/blob/master/test/test_ops.py#L285-L299), outputs and gradients of `fn`(which is `op.output_func ∘ op.get_op`) are compared to the outputs and gradients of `script_fn` (which is  `op.output_func ∘ op.op`). Those are different functions if custom `op.get_op` is used and can't be compared.
Outputs and grads are compared [here.](https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/common_jit.py#L81-L82)

Reply:
You're right. I'll file an issue for this. Seems like we will need to add an input_func.